pickhardtpayments/Attempt.py

```diff
@@ -0,0 +1,120 @@
+from enum import Enum
+
+import UncertaintyChannel
+
+
+class AttemptStatus(Enum):
+    PLANNED = 1
+    INFLIGHT = 2
+    ARRIVED = 3
+    FAILED = 4
+    SETTLED = 5
+
+
+class Attempt:
+    """
+    An Attempt describes a path (a set of channels) of an amount from sender to receiver.
+
+    When sending an amount of sats from sender to receiver, a payment is usually split up and sent across
+    several paths, to increase the probability of being successfully delivered. Each of this path is referred to as an
+    Attempt.
+    An Attempt consists of a list of Channels (class:Channel) and the amount in sats to be sent through this path.
+
+    :param path: a list of UncertaintyChannel objects from sender to receiver
+    :type path: list[UncertaintyChannel]
+    :param amount: the amount to be transferred from source to destination
+    :type amount: int
+    """
+
+    def __init__(self, path: list[UncertaintyChannel], amount: int = 0):
+        """Constructor method
+        """
+        self._routing_fee = -1
```

Comment:
yes, negative routing fees will indeed break the -1 idea. I only changed it in payment, now changed it in attempt as well. 
I would actually extract the `planned`/`inflight` problem into its own issue. i fear that discussion around it will get lost in this Thread.